src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -545,7 +545,11 @@ type DiagnosticItem =
             | ErrorCode.ExpectingIterableExpr                     -> "The type {0} does not support iteration. Expecting an expression of array type or of type Range."
             | ErrorCode.ExpectingCallableExpr                     -> "The type of the expression must be a function or operation type. The given expression is of type {0}." 
             | ErrorCode.UnknownIdentifier                         -> "No identifier with the name \"{0}\" exists."
-            | ErrorCode.UnsupportedResultComparison               -> "Measurement results cannot be compared because the execution target supports only {0} runtime capabilities."
+            | ErrorCode.UnsupportedResultComparison               ->
+                // TODO: When the names of the runtime capabilities are finalized, they can be included in the error
+                // message.
+                "The execution target {0} does not support comparing measurement results. " +
+                "Choose an execution target with additional capabilities or use a controlled operation instead."
```

Comment:
I would get rid of this second half of the message - using controlled (a coherent if) is just usually not used with the same purpose as branching based on measurement result, and the suggestion hence seems odd to me.